src/run_info.c

```diff
@@ -203,7 +201,7 @@ void RAI_FreeRunInfo(RedisModuleCtx *ctx, struct RedisAI_RunInfo *rinfo) {
     if (!rinfo) {
         return;
     }
-    if (rinfo->master == 0) {
+    if (*rinfo->dagRefCount == 0) {
```

Comment:
Isn't this the exact opposite?

Reply:
you are right of course....
Are we testing it somewhere? This should have crash for at least two rinfo copies right? but the tests passed....

Reply:
We are testing DAGs on multiple devices, but whether they would crash if deallocated out of order it needs to be investigated. It might be that they crash once every 20 runs.